src/primitives/transaction.cpp

```diff
@@ -47,8 +48,19 @@ static uint256 ComputeCMutableTransactionHash(const CMutableTransaction &tx) {
     return SerializeHash(tx, SER_GETHASH, 0);
 }
 
+static uint256 ComputeTxId(int32_t nVersion, const std::vector<CTxIn> &vin,
+                           const std::vector<CTxOut> &vout,
+                           uint32_t nLockTime) {
+    CHashWriter txid(SER_GETHASH, 0);
+    txid << nVersion;
+    txid << TxInputsMerkleRoot(vin);
```

Comment:
Is it at all useful to commit to the number of inputs and outputs?

Reply:
We need the height of the tree committed to here so that you can't provide false proofs of heigher order than what is in the TxID.

Reply:
I would put a comment about why the height is committed to so that people know why and its clear that contracts need to validate it.